The path of live os image in oe is hardcoded in anaconda-init.

Upstream-Status: Inappropriate [oe specific]

Signed-off-by: Kai Kang <kai.kang@windriver.com>

Rebase for anaconda 37.

Signed-off-by: Kai Kang <kai.kang@windriver.com>

Add mount option 'noload' to resolve mount error for F39.

Signed-off-by: Kai Kang <kai.kang@windriver.com>

---
 pyanaconda/modules/payloads/source/live_os/initialization.py | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/pyanaconda/modules/payloads/source/live_os/initialization.py b/pyanaconda/modules/payloads/source/live_os/initialization.py
index 50de78a087..02656e3da7 100644
--- a/pyanaconda/modules/payloads/source/live_os/initialization.py
+++ b/pyanaconda/modules/payloads/source/live_os/initialization.py
@@ -49,10 +49,7 @@ class DetectLiveOSImageTask(Task):
 
         :return: a path of a block device or None
         """
-        block_device = \
-            self._check_block_device("/dev/mapper/live-base") or \
-            self._check_block_device("/dev/mapper/live-osimg-min") or \
-            self._check_mount_point("/run/rootfsbase")
+        block_device = '/dev/loop3'
 
         if not block_device:
             raise SourceSetupError("No Live OS image found!")
@@ -157,7 +154,7 @@ class SetUpLiveOSSourceTask(SetUpMountTask):
                 device_path,
                 self._target_mount,
                 fstype="auto",
-                options="ro"
+                options="ro,noload"
             )
         except OSError as e:
             raise SourceSetupError(str(e)) from e
